bluesky/run_engine.py

```diff
@@ -841,11 +877,25 @@ def _save(self, msg):
         if objs_read not in self._descriptor_uids:
             # We don't not have an Event Descriptor for this set.
             data_keys = {}
-            [data_keys.update(self._describe_cache[obj]) for obj in objs_read]
+            configuration = {}
+            configuration_times = {}
+            for obj in objs_read:
+                dks = self._describe_cache[obj]
+                # Mark each data_key with a 'group' ID which will be matched
+                # to the object's configuration, shared by all obj's data keys.
+                group = str(hash(frozenset(dks)))  # unique id for this object
+                for field, dk in dks.items():
+                    dk['group'] = group
+                data_keys.update(dks)
+                configuration[group] = self._configuration_cache[obj]
+                configuration_times[group] = \
+                    self._configuration_times_cache[obj]
             _fill_missing_fields(data_keys)  # TODO Move this to ophyd/controls
             descriptor_uid = new_uid()
             doc = dict(run_start=self._run_start_uid, time=ttime.time(),
-                       data_keys=data_keys, uid=descriptor_uid)
+                       data_keys=data_keys, uid=descriptor_uid,
```

Comment:
@arkilic @klauer @dchabot 

This is the interesting bit of this PR in that it adds a path to add configuration information into mds.  The basic scheme is that each data key picks up another entry `'group'` and there is a `'configuration'` and `'configuration_times'` are dicts keyed on those groups which store what ever that object thought was important to store.

@danielballan is working on code to 'pivot' any thing in the configuration dict up into the events (ex, every event would have an entry with the exposure time).